cmd/agent/main.go

```diff
@@ -14,10 +14,11 @@ import (
 func main() {
 	agentCfg, err := config.NewAgentConfig()
 	if err != nil {
-		log.Fatal("failed to get agent configurations: " + err.Error())
+		agentCfg.Log.ErrorContext(context.Background(), "failed to get agent configurations: "+err.Error())
```

Comment:
Check out a better way to log errors: https://stackoverflow.com/questions/77304845/how-to-log-errors-with-log-slog